mergify_engine/queue/merge_train.py

```diff
@@ -891,9 +917,26 @@ async def update_summaries(
                     f"The pull requests {refs} cannot be merged and will be split"
                 )
 
-        checks_timeout_summary = (
-            "\n\n⏲️  The checks have timed out ⏲️" if self.has_timed_out else ""
-        )
+        if (
+            self.checks_timeout is not None
+            and self.checks_conclusion != check_api.Conclusion.SUCCESS
+            and self.last_evaluated_conditions is not None
+        ):
+            expected_finish_timestamp = self.creation_date + self.checks_timeout
+            expected_finish = (
+                date.pretty_datetime(
+                    date.RelativeDatetime(expected_finish_timestamp).value
+                )
+                if expected_finish_timestamp.date() > date.utcnow().date()
+                else date.pretty_time(
+                    date.RelativeDatetime(expected_finish_timestamp).value
+                )
+            )
+            self.last_evaluated_conditions += (
```

Comment:
We should avoid setting TrainCar state attributes in update_summary(), if we need to update the state we should use update_state().

Reply:
`self.last_evaluated_conditions` is no longer modified now !